evm/src/main/java/org/hyperledger/besu/evm/precompile/ECRECPrecompiledContract.java

```diff
@@ -88,20 +87,22 @@ public PrecompileContractResult computePrecompile(
     }
 
     PrecompileInputResultTuple res;
-
+    Integer cacheKey = null;
     if (enableResultCaching) {
-      res = ecrecCache.getIfPresent(nonMutatedInput.hashCode());
+      cacheKey = input.hashCode();
```

Comment:
can we use getCacheKey here ? why using hashcode instead of Arrays.hashCode(input.toArrayUnsafe());

Reply:
ECREC was the first precompile "hash collision" I looked into, hashCode was appropriate for it.  But I agree there is no reason not to use the Arrays.hashCode for it also